src/uwtools/tests/test_cli.py

```diff
@@ -234,62 +351,28 @@ def test_main_raises_exception(capsys):
 
 
 def test__parse_args():
-    raw_args = ["foo", "--bar", "88"]
+    raw_args = ["testing", "--bar", "88"]
     with patch.object(cli, "Parser") as Parser:
         cli._parse_args(raw_args)
         Parser.assert_called_once()
         parser = Parser()
         parser.parse_args.assert_called_with(raw_args)
 
 
-@pytest.mark.parametrize(
-    "vals",
-    [
-        (ns(file_1_path=None, file_1_format=None), "--file-1-path", "--file-1-format"),
-        (ns(file_2_path=None, file_2_format=None), "--file-2-path", "--file-2-format"),
-        (ns(input_file=None, input_format=None), "--input-file", "--input-format"),
-        (ns(output_file=None, output_format=None), "--output-file", "--output-format"),
-        (ns(values_file=None, values_format=None), "--values-file", "--values-format"),
-    ],
-)
-def test__set_formats_fail(capsys, vals):
-    # When reading/writing from/to stdin/stdout, the data format must be specified, since there is
-    # no filename to deduce it from.
-    args, path_arg, fmt_arg = vals
-    with raises(SystemExit):
-        cli._set_formats(args)
-    assert f"Specify {fmt_arg} when {path_arg} is not specified" in capsys.readouterr().err
-
-
-def test__set_formats_pass_explicit():
-    # Accept explcitly-specified format, whatever it is.
-    args = ns(input_file="/path/to/input.txt", input_format="jpg")
-    args = cli._set_formats(args)
-    assert args.input_format == "jpg"
-
-
-@pytest.mark.parametrize("fmt", vars(FORMAT).keys())
-def test__set_formats_pass_implicit(fmt):
-    # The format is correctly deduced for a file with a known extension.
-    args = ns(input_file=f"/path/to/input.{fmt}", input_format=None)
-    args = cli._set_formats(args)
-    assert args.input_format == vars(FORMAT)[fmt]
-
-
 # Helper functions
 
 
-@fixture
-def subparsers():
-    # Create and return a subparsers test object.
-    return Parser().add_subparsers()
-
-
 def submodes(parser: Parser) -> List[str]:
     # Return submodes (named subparsers) belonging to the given parser.
     if subparsers := parser._subparsers:
-        if action := subparsers._actions[3]:
+        if action := subparsers._actions[1]:
```

Comment:
Hmmmm. This change seems a little magical.

Reply:
It's awful. I haven't found a more robust way to introspect the subparser configuration. Using a private attribute is bad enough, but the fact that the location of the submodes in the `subparsers` object is apparently sensitive to the order in which arguments are added is even worse. I'll be on the lookout for something better, but this was all I was able to find so far.

Reply:
Would a comment here offer any insight on what the number is for? Or where to find the code that would explain it?

Reply:
Ugh, its like inspect.stack(), isn't it? I don't know any better way, myself. Might be able to drop a small comment?

Reply:
I'll add a comment so we at least know what it means, and where to look if something changes.

Reply:
I found a way to at least get rid of the magic number by filtering out what we need. It's still ugly, as `argparse` wasn't designed for this kind of introspection, but its a little shorter. I've also updated the comment to link to a StackOverflow article where someone asked about this.